tests/mock_network/drop.rs

```diff
@@ -14,19 +14,11 @@ use routing::{event::Event, mock::Environment, NetworkParams};
 
 // Drop node at index and verify its own section detected it.
 fn drop_node(nodes: &mut Vec<TestNode>, index: usize) {
-    let mut node = nodes.remove(index);
-    let close_names = node.close_names();
-    let endpoint = node.endpoint();
-
-    drop(node);
+    let _ = nodes.remove(index);
 
     // Using poll_all instead of poll_and_resend here to only let the other nodes realise the node
     // got disconnected, but not make more progress.
     let _ = poll_all(nodes);
-
-    for node in nodes.iter_mut().filter(|n| close_names.contains(&n.name())) {
-        assert!(!node.inner.is_connected(&endpoint));
```

Comment:
Did you solve the detecting offline node then? because we only get an offline event if we tried to connect?
That would be great news.

Reply:
Not sure what you mean?

Reply:
How does offline detection works now?for elders? for adults?

Reply:
By receiving the `ConnectionFailure` event.

Reply:
So I guess we are still connected or trying to send messages then, because otherwise why would we get a connection failure? Will this happen on all Elders?

Reply:
I'm wondering, if adults are not connected to all elders currently, would they detect them dropping before an Elder change occur? Is there a missing test for that?

Reply:
> I'm wondering, if adults are not connected to all elders currently, would they detect them dropping before an Elder change occur?

I don't think so, but why would adults need to be able to detect elder going offline? We currently don't do anything there: https://github.com/maidsafe/routing/blob/fleming/src/states/adult/mod.rs#L439-L442 

> Is there a missing test for that?

As adults don't do anything on peer drop I'm not sure what is there to test. Is there maybe an edge case you see?